### ISSUE TITLE
callRemote exception handler relies on possibly unbound local `req'

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#10363]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#459|
|Type|defect|
|Created|2004-01-03 14:37:41Z|

<details><summary>Searchable metadata</summary>

```
trac-id__459 459
type__defect defect
reporter__exarkun exarkun
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__pb pb
keywords__ 
time__1073140661000000 1073140661000000
changetime__1470831661090355 1470831661090355
version__ 
owner__warner warner
cc__exarkun cc__warner
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

```
#!html
<pre>
In pb.RemoteReference.callRemote (sandbox), If self.broker.newRequestID() raises
an exception, the failure cannot be reported because the PendingRequest is never
instantiated.  Instantiating it before doing anything should remove the
possibility of a problem.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="warner's avatar" src="https://avatars.githubusercontent.com/u/27146?s=50" width="50" height="50">](https://github.com/warner)|@warner commented|
|-|-|

```
#!html
<pre>
this has been fixed. If newRequestID fails, a new Deferred is created,
errbacked, and returned.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|GitHub commented|
|-|-|

In changeset e16a6e2df9a240be2245cc84e10667b57b087759
```
#!CommitTicketReference repository="" revision="e16a6e2df9a240be2245cc84e10667b57b087759"
Merge pull request #459 from twisted/secure-session-3461-2

Author: steiza, glyph

Reviewer: glyph, exarkun, Lukasa

Fixes: #3461 

Stop sending secure session cookies over unsecured connections.
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

In changeset 0d44542bbafa7bb7f139edeb8c42b07a9c3bf9d0
```
#!CommitTicketReference repository="" revision="0d44542bbafa7bb7f139edeb8c42b07a9c3bf9d0"
Merge pull request #459 from twisted/secure-session-3461-2

Author: steiza, glyph

Reviewer: glyph, exarkun, Lukasa

Fixes: #3461 

Stop sending secure session cookies over unsecured connections.
```
